Sources/Orbit/Components/ListChoice.swift

```diff
@@ -0,0 +1,252 @@
+import SwiftUI
+
+/// Shows one of a selectable list of items with similar structures.
+///
+/// - Note: [Orbit definition](https://orbit.kiwi/components/listchoice/)
+/// - Important: Component expands horizontally to infinity up to a ``Layout/readableMaxWidth``.
+public struct ListChoice: View {
+
+    let title: String
+    let description: String
+    let icon: Icon.Symbol
+    let trailingElement: ListChoice.TrailingElement
+    let backgroundColor: BackgroundColor?
+    let action: () -> Void
+
+    public var body: some View {
+        SwiftUI.Button(
+            action: {
+                HapticsProvider.sendHapticFeedback(.light(0.5))
+                action()
+            },
+            label: {
+                buttonContent
+            }
+        )
+        .buttonStyle(ListChoiceButtonStyle(backgroundColor: backgroundColor))
+        .accessibility(label: SwiftUI.Text(title))
+        .accessibility(hint: SwiftUI.Text(description))
+    }
+
+    @ViewBuilder var buttonContent: some View {
+        HStack(spacing: 0) {
+            header
+                .frame(maxWidth: .infinity, alignment: .leading)
+
+            // Provide minimal height (frame(minHeight:) collapses multiline text in snapshots)
+            Color.clear
+                .frame(width: 0, height: 48)
+
+            trailingElementView
+                .padding(.trailing, .medium)
+        }
+        .frame(maxWidth: Layout.readableMaxWidth, alignment: .leading)
+        .overlay(separator, alignment: .bottom)
+    }
+    
+    @ViewBuilder var header: some View {
+        if isHeaderEmpty == false {
+            HStack(alignment: .firstTextBaseline, spacing: .small) {
+                iconView
+                    .alignmentGuide(.firstTextBaseline) { _ in
+                        Heading.Style.title3.size * 1.1
+                    }
+
+                HStack(alignment: .firstTextBaseline, spacing: .xxSmall) {
+                    VStack(alignment: .leading, spacing: .xxxSmall) {
+                        Text(title)
+                        Text(description, size: .small, color: .inkLight)
+                    }
+                    .padding(.vertical, headerTextPadding)
+
+                    Spacer(minLength: 0)
+                }
+            }
+            .padding(.horizontal, .medium)
+        }
+    }
+
+    @ViewBuilder var trailingElementView: some View {
+        switch trailingElement {
+            case .none:
+                EmptyView()
+            case .disclosure:
+                Icon(symbol: .chevronRight, size: .medium, color: .inkNormal)
+            case .button(let type, let action):
+                switch type {
+                    case .add:
+                        Button(.plus, style: .primarySubtle, size: .small, action: action)
```

Comment:
Buttons will need their own vertical .small padding (12) to increase the height of the whole component. As this will make the switch cases more duplicated, it could be extracted to `@ViewBuilder func button(type:action:) -> some View`